aten/src/ATen/core/ivalue_inl.h

```diff
@@ -658,11 +659,14 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
       event.block(impl_.getStream(event.device()));
     }
 
-    for (const c10::Storage& storage : storages_) {
-      const at::DataPtr& data_ptr = storage.data_ptr();
-      if (!data_ptr.device().is_cpu()) {
+    for (const WeakStorage& weak_storage : storages_) {
+      c10::intrusive_ptr<c10::StorageImpl> storage = weak_storage.lock();
+      if (!storage) {
+        continue;
```

Comment:
do we need any warning here?

Reply:
At this point I think if we did it would fire all the time. Hence for the moment I'm perhaps more inclined to keep this silent: this way we don't introduce any (visible) functional change wrt the previous code, the only difference being that Future doesn't cause any memory to "leak".